In [3]:
import pandas as pd
df = pd.read_csv('../datasets/twitter_training.csv', header=None, names=['ID', 'Entity', 'Sentiment', 'Tweet'])
print(df.head(100))

      ID       Entity Sentiment  \
0   2401  Borderlands  Positive   
1   2401  Borderlands  Positive   
2   2401  Borderlands  Positive   
3   2401  Borderlands  Positive   
4   2401  Borderlands  Positive   
..   ...          ...       ...   
95  2416  Borderlands   Neutral   
96  2417  Borderlands  Negative   
97  2417  Borderlands  Negative   
98  2417  Borderlands  Negative   
99  2417  Borderlands  Negative   

                                                Tweet  
0   im getting on borderlands and i will murder yo...  
1   I am coming to the borders and I will kill you...  
2   im getting on borderlands and i will kill you ...  
3   im coming on borderlands and i will murder you...  
4   im getting on borderlands 2 and i will murder ...  
..                                                ...  
95  I hate that this easy horrible modifier event ...  
96  Grounded almost looked pretty cool even despit...  
97  Grounded looked cool despite the borderline un...  
98  Grosskreutz loo

## Prompt
Quiero que primero analices el prompt que te compartiré, y me digas si le falta algo en estructura KERNEL, no quiero que tenga indicaciones basura y sea óptimo.


"
De acuerdo a los datos del dataframe df, quiero lo siguiente:
Eres un experto en data science en identificación de patrones. Y debes analizar los tweets que te comparta. Debes clasificar el texto de acuerdo a 3 categorías: Positivo, Negativo y Neutral.
El formato de respuesta debe ser en Json con llave id y sentimiento, solamente me deberás dar la clasificación sin explicación.
Si solamente son caracteres especiales o no tiene sentido, clasificaclo como irrelevante.
"

## Respuesta

"""

K - Knowledge (Contexto)
Tienes un DataFrame llamado 'df' con las columnas: ID, Entity, Sentiment, Tweet.
Tu tarea es reclasificar la columna 'Tweet' basándote únicamente en el contenido del texto.

E - Expectations (Expectativas)
Clasifica cada tweet en UNA de estas categorías:
- Positive
- Negative  
- Neutral
- Irrelevant (si solo contiene caracteres especiales o texto sin sentido)

R - Role (Rol)
Actúa como experto en análisis de sentimientos con experiencia en procesamiento de lenguaje natural y clasificación de texto en redes sociales.

N - Narrowing (Acotación)
Reglas estrictas:
1. Analiza SOLO el contenido de la columna 'Tweet'
2. Ignora la columna 'Sentiment' existente (es para validación posterior)
3. NO proporciones explicaciones ni razonamientos
4. Clasifica tweets vacíos o con solo símbolos como "Irrelevant"

E - Expectations (Formato de salida)
Para cada tweet, devuelve UN objeto JSON con esta estructura exacta:
{"id": <valor_de_columna_ID>, "sentiment": "<categoría>"}

Ejemplo de salida esperada:
{"id": 2401, "sentiment": "Positive"}
{"id": 2402, "sentiment": "Irrelevant"}

L - Limitations (Limitaciones)
- Procesa los tweets de uno en uno cuando se te solicite
- Usa exactamente los nombres de categorías especificados (case-sensitive)
- NO agregues campos adicionales al JSON

"""

## Response

Voy a crear una lista de tareas para rastrear el trabajo (qué voy a hacer y en qué orden), luego clasificaré cada tweet y devolveré los JSONs solicitados. Resultado: lista creada y luego produciré las salidas JSON para cada tweet.

Created 7 todos

Progreso: lista de tareas creada y requisitos confirmados. Ahora devuelvo los JSONs con la clasificación (sin explicaciones), uno por línea.

{"id": 4433, "sentiment": "Negative"}
{"id": 6273, "sentiment": "Negative"}
{"id": 7925, "sentiment": "Positive"}
{"id": 11332, "sentiment": "Positive"}
{"id": 1107, "sentiment": "Positive"}